File: jeonghyeon/4week/로그인 성공.js
function solution(id_pw, db) {
    const memberinfo = db.find((member) => member[0] === id_pw[0]);
    // find() : 배열의 요소를 순차적으로 순회하면서 조건에 일치하는 요소의 값 반환
    // 일치하는 회원 정보 memberinfo에 저장
    if(!memberinfo) { // 아이디 불일치
        return "fail";
    }
    if(memberinfo[1] === id_pw[1]){ 
        // 아이디 일치 후 등록 되어있는 비밀번호랑 입력된 비밀번호 비교
       return "login"; // 같을 시 "login"
    } else { // 비밀번호 다를 경우
        return "wrong pw"; // "wrong pw"
    }
}

File: dongmin/week4/로그인 성공.java
import java.util.HashMap;

class Solution {
    public String solution(String[] id_pw, String[][] db) {
        HashMap<String, String> map = new HashMap<String, String>();    //HashMap은 키와 값의 형태를 띄움.
        for (String[] strings : db) {
            map.put(strings[0],strings[1]);     //아이디 비밀번호를 아이디를 키로 비밀번호를 값으로 등록한다.
        }

        if(map.containsKey(id_pw[0])) {  //맵에 key값(아이디)이 있는지 확인 가능
            if (map.get(id_pw[0]).equals(id_pw[1]))  //map에 매핑된 비밀번호와 id_pw 배열에 저장되어있는 비밀번호를 비교
                return "login";
            else
                return "wrong pw";
        } else return "fail";
        
    }
}